### DISCUSSION TITLE
Add polyfill for `ext-ds`

### DISCUSSION THREAD

**Discussion by llaville** [composer-unused/composer-unused#506] [Ideas]
Hello,

As you used [`ext-ds`](https://pecl.php.net/package/ds), as development requirement, since relase [0.7.4](https://github.com/composer-unused/composer-unused/releases/tag/0.7.4) , I suggest to add the [Polyfill](https://github.com/php-ds/polyfill).

```json
    "require-dev": {
        "ext-ds": "*",
        "php-ds/php-ds": "^1.4",
    },
```
If extension is available, the polyfill won't be used, and if polyfill is installed, that will avoid following error during installation !

This will enhance User Experience, rather than to get, when running `composer install` (because the lock file included the extension)

![ext-ds_missing](https://user-images.githubusercontent.com/364342/236602571-bcff10fc-2a5e-4fc3-b9a6-38ea8f195a7a.png)

Even, if we can avoid this with Composer tip 

```
Alternatively, you can run Composer with `--ignore-platform-req=ext-ds` to temporarily ignore these required extensions.
```

What do you think ?

**Comment by icanhazstring** [comment]
Hi. This sounds like a nice idea.
If you would provide a PR I will convert it to an issue to be tracked 🙂

Thanks 💪

// See #507
